packages/ember-data/lib/adapters/rest_adapter.js

```diff
@@ -171,6 +171,58 @@ export default Adapter.extend({
   defaultSerializer: '-rest',
 
   /**
+    By default, the RESTAdapter will send the query params sorted alphabetically to the
+    server.
+
+    For example:
+    ```js
+      store.find('posts', {sort: 'price', category: 'pets'});
+    ```
+    will generate a requests like this `/posts?category=pets&sort=price`, even if the
+    parameters were specified in a different order.
+
+    That way the generated URL will be deterministic and that simplifies caching mechanisms
+    in the backend.
+
+    Setting `sortQueryParams` to a falsey value will respect the original order.
+
+    In case you want to sort the query parameters with a different criteria, set
+    `sortQueryParams` to your custom sort function.
+
+    ```js
+    export default DS.RESTAdapter.extend({
+      sortQueryParams: function(params) {
+        var sortedKeys = Object.keys(params).sort().reverse();
+        var len = sortedKeys.length, newParams = {};
+
+        for (var i = 0; i < len; i++) {
+          newParams[sortedKeys[i]] = params[sortedKeys[i]];
+        }
+        return newParams;
+      }
+    });
+    ```
+
+    @method sortQueryParams
+    @param {Object} obj
+    @return {Object}
+  */
+  sortQueryParams: function(obj) {
+    var keys = Object.keys(obj);
```

Comment:
Lets use `Ember.keys` here to support older browsers that don't have `Object.keys`.